djstripe/models/payment_methods.py

```diff
@@ -157,20 +191,35 @@ def api_retrieve(self, api_key=None, stripe_account=None):
         # either a customer or account is supplied to determine the
         # correct object to use.
         api_key = api_key or self.default_api_key
-        customer = self.customer.api_retrieve(
-            api_key=api_key, stripe_account=stripe_account
-        )
 
-        # If the customer is deleted, the sources attribute will be absent.
-        # eg. {"id": "cus_XXXXXXXX", "deleted": True}
-        if "sources" not in customer:
-            # We fake a native stripe InvalidRequestError so that it's caught
-            # like an invalid ID error.
-            raise InvalidRequestError(f"No such source: {self.id!r}", "id")
+        if self.customer:
+            customer = self.customer.api_retrieve(
+                api_key=api_key, stripe_account=stripe_account
+            )
+
+            # If the customer is deleted, the sources attribute will be absent.
+            # eg. {"id": "cus_XXXXXXXX", "deleted": True}
+            if "sources" not in customer:
+                # We fake a native stripe InvalidRequestError so that it's caught
+                # like an invalid ID error.
+                raise InvalidRequestError(f"No such source: {self.id!r}", "id")
+
+            # This will retrieve the source using the account ID where the customer resides,
+            # so we don't have to pass `stripe_account`.
+            return customer.sources.retrieve(self.id, expand=self.expand_fields)
+        if self.account:
+            account = self.account.api_retrieve(
+                api_key=api_key, stripe_account=stripe_account
+            )
+
+            # TODO Handle the case the Stripe Connected Account has no External Accounts
+            # # populated
+            # if "external_accounts" not in account:
+            #     pass
 
-        # This will retrieve the source using the account ID where the customer resides,
-        # so we don't have to pass `stripe_account`.
-        return customer.sources.retrieve(self.id, expand=self.expand_fields)
+            # This will retrieve the external_accounts using the bank_account ID where the account resides,
+            # so we don't have to pass `stripe_account`.
```

Comment:
Is dropping expand_fields going to be a problem here?

Reply:
I didn't write this line: ```customer.sources.retrieve(self.id, expand=self.expand_fields)``` But looking at the code, I think ```expand=self.expand_fields``` can be dropped without any problem. This is because ```self.customer.api_retrieve(
                api_key=api_key, stripe_account=stripe_account
            )``` will use `expand_fields` anyway.